doc/BGP/BGP-router-id.md

```diff
@@ -0,0 +1,92 @@
+# BGP Router ID Customized
+
+- [Revision](#revision)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Scope](#scope)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High Level Design](#high-level-design)
+- [Config DB Enhancement](#config-db-enhancement)
+    - [DEVICE_METADATA](#device_metadata)
+
+### Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Mar 27 2024 | Yaqiang Zhu, Jing Kan | Initial proposal   |
+
+### Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+| ------------------------ | ----------- |
+| FRR | A free and open source Internet routing protocol suite for Linux and Unix platforms |
+| Router ID | 32-bit value that uniquely identifies a BGP device |
+
+### Scope
+
+This document describes a mechanism that to use customized router id when Loopback IPv4 address is not configured.
```

Comment:
This statement is not aligned with #2 item in `High Level Design`.

This design is to allow user to explicitly configure BGP router id.